src/document/position.js

```diff
@@ -97,13 +119,13 @@ CKEDITOR.define( [ 'utils', 'ckeditorerror' ], ( utils, CKEditorError ) => {
 		}
 
 		/**
-		 * Two positions equal if paths are equal.
+		 * Two positions equal if paths are equal and roots are the same.
 		 *
 		 * @param {document.Position} otherPosition Position to compare.
 		 * @returns {Boolean} True if positions equal.
 		 */
 		isEqual( otherPosition ) {
```

Comment:
This is a correct observation. BTW. I am fine with `utils.compareArrays` flags (`SAME`, `EXTENSION`, `PREFIX`) because you have to require `utils` anyway to use `compareArrays`.